### PR TITLE
feat: implementing prune

### PR DESCRIPTION
**Pull request by Farenheith**
The prune method clear expired values and remove totally expired branches, keeping the tree the compactest as possible

### CODE REVIEW

**Review by gustavo-rodrigues-dev — DISMISSED** [review]

**Review by DanielM08 — DISMISSED** [review]

**Review by DanielM08 — APPROVED** [review]

File: src/tree-key-cache.ts [hunk 1]
```diff
@@ -285,17 +285,15 @@ export class TreeKeyCache<
...
+					if (nodeRef) {
 						nodeRef = createTraversalItem(
 							nodeRef.key,
 							nodeRef.level,
 							nodeRef.parentRef,
 							tree,
-							tree[TreeKeys.value],
+							!isUndefined(value) && this.isNotExpired(deadline, now)
+								? tree[TreeKeys.value]
+								: undefined,
```
Comment (Farenheith) [inline src/tree-key-cache.ts:296]: Nodes with empty value weren't being returned correctly, even though it can have children. that's why this change was made

**Review by pedrosodre — APPROVED** [review]
